cnd/src/network.rs

```diff
@@ -301,75 +289,58 @@ impl ComitNode {
 
     pub fn initiate_communication(
         &mut self,
-        id: LocalSwapId,
-        swap_params: Herc20HalightBitcoinCreateSwapParams,
-    ) -> anyhow::Result<()> {
-        self.supports_halight()?;
-        self.comit.initiate_communication(id, swap_params)
-    }
-
-    fn init_hbit_herc20(
-        &mut self,
-        id: LocalSwapId,
-        swap: CreatedSwap<hbit::CreatedSwap, herc20::CreatedSwap>,
-    ) -> anyhow::Result<()> {
-        self.comit.init_hbit_herc20(id, swap)
-    }
-
-    fn init_herc20_hbit(
-        &mut self,
-        id: LocalSwapId,
-        swap: CreatedSwap<herc20::CreatedSwap, hbit::CreatedSwap>,
+        local_swap_id: LocalSwapId,
+        peer: DialInformation,
+        role: Role,
+        digest: SwapDigest,
+        identities: Identities,
     ) -> anyhow::Result<()> {
-        self.comit.init_herc20_hbit(id, swap)
-    }
-
-    pub fn get_created_swap(
-        &mut self,
-        id: LocalSwapId,
-    ) -> Option<Herc20HalightBitcoinCreateSwapParams> {
-        self.comit.get_created_swap(&id)
-    }
+        // At this stage we do not know if the arguments passed to us make up a
```

Comment:
Why not trusting the controller?

Reply:
Because we make no assertions on the data received by the controller from the user, we _definitely_ don't trust the user do we?

Reply:
Shouldn't the controller do the assertion?